src/pairs/mento-v2.ts

```diff
@@ -0,0 +1,258 @@
+  IBiPoolManager,
+  ISortedOracles__factory,
+} from "@mento-protocol/mento-core-ts";
+import { BigNumber } from "bignumber.js";
+import { ethers } from "ethers";
+
+import { Address, Pair, Snapshot, BigNumberString } from "../pair";
+import { selectAddress } from "../utils";
+// these are just placeholder addresses I've added, pairmentov2 has not been deployed yet
+import { address as mainnetPairMentoV2Address } from "../../tools/deployed/mainnet.PairMentoV2.addr.json";
+enum PricingFunctionType {
+  ConstantProduct = "ConstantProduct",
+  ConstantSum = "ConstantSum",
+}
+
+const FIXED1 = new BigNumber(1000000000000000000000000); // 10^24
+interface PairMentoV2Snapshot extends Snapshot {
+  swapFee: BigNumberString;
+  updateFrequency: BigNumberString;
+  pricingModule: PricingFunctionType;
+  bucket0: BigNumberString;
+  bucket1: BigNumberString;
+}
+
+export class PairMentoV2 extends Pair {
+  private poolExchange!: IBiPoolManager.PoolExchangeStructOutput;
+  private swapFee: BigNumber = new BigNumber(0);
+  private updateFrequency: BigNumber = new BigNumber(0);
+  private pricingModule!: PricingFunctionType;
+  private bucket0: BigNumber = new BigNumber(0);
+  private bucket1: BigNumber = new BigNumber(0);
+
+  constructor(
+    chainId: number,
+    private provider: ethers.providers.Provider,
+    private mento: Mento,
+    private exchange: Exchange,
+    private sortedOraclesAddress: string
+  ) {
+    super(
+      selectAddress(chainId, {
+        mainnet: mainnetPairMentoV2Address,
+        }
+      )
+    );
+  }
+
+  protected async _init(): Promise<{
+    pairKey: string | null;
+    tokenA: string;
+    tokenB: string;
+  }> {
+    this.poolExchange = await this.getPoolExchange();
+    this.pricingModule = await this.getPricingModuleName(
+      this.poolExchange.pricingModule
+    );
+    this.swapFee =  new BigNumber(this.poolExchange.config.spread.value._hex);
+    this.updateFrequency = new BigNumber(this.poolExchange.config.referenceRateResetFrequency._hex);
+    return {
+      pairKey: this.exchange.id,
+      tokenA: this.exchange.assets[0],
+      tokenB: this.exchange.assets[1],
+    };
+  }
+
+  public async refresh() {
+    const [lastBucketUpdate] = [
+      new BigNumber(this.poolExchange.lastBucketUpdate._hex),
+    ];
+    const tillUpdateSecs = lastBucketUpdate
+      .plus(this.updateFrequency)
+      .minus(Date.now() / 1000);
+    let buckets: { bucket0: BigNumber; bucket1: BigNumber };
+    if (tillUpdateSecs.lte(5)) {
+      buckets = await this.mentoBucketsAfterUpdate();
+    } else {
+      const [bucket0, bucket1] = [
+        new BigNumber(this.poolExchange.bucket0._hex),
+        new BigNumber(this.poolExchange.bucket1._hex),
+      ];
+      buckets = { bucket0, bucket1 };
+    }
+    this.bucket0 = buckets.bucket0
+    this.bucket1 = buckets.bucket1
+  }
+
+  public swapExtraData(): string {
+    const broker = this.mento.getBroker();
+    return `${broker.address}${this.exchange.providerAddr.replace(
```

Comment:
instead of replace, substring(2) would probably be more correct right?

Reply:
yes, makes sense, substring() is cleaner